### PR TITLE
feat: react plugin instead of old angular thing

### LINKED ISSUE #4: Update from Angular to React
While #3 is fixed for now, eventually we need to update the whole thing to react.

See also: https://corpglory.com/s/grafana-react-plugins/

### PR DESCRIPTION
**Pull request by kinnarr**
Fixes #4 

- [x] update readme
- [x] more testing
- [x] test existing panels

### CODE REVIEW

File: src/QueryEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,158 @@
+import defaults from 'lodash/defaults';
+
+import React, { ChangeEvent, PureComponent } from 'react';
+import { AsyncSelect, InlineField, InlineFieldRow, InlineSwitch, Input, MultiSelect } from '@grafana/ui';
+import { QueryEditorProps, SelectableValue } from '@grafana/data';
+import { MetricQDatasource } from './datasource';
+import { defaultQuery, MetricQDataSourceOptions, MetricQQuery } from './types';
+
+type Props = QueryEditorProps<MetricQDatasource, MetricQQuery, MetricQDataSourceOptions>;
+export interface State {
```
Comment (tilsche) [inline src/QueryEditor.tsx:10]: can we remove `export`?

File: src/QueryEditor.tsx [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+          </InlineField>
+        </InlineFieldRow>
+        <InlineFieldRow>
+          <InlineField label="Name" grow={true} labelWidth={labelWidth}>
+            <Input value={name} onChange={this.onNameChange} placeholder={'$metric/$function'} />
+          </InlineField>
+        </InlineFieldRow>
+        <InlineFieldRow>
+          <InlineField label="Functions" grow={true} labelWidth={labelWidth}>
+            <MultiSelect
```
Comment (tilsche) [inline src/QueryEditor.tsx:129]: Add string type?

File: src/QueryEditor.tsx [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    };
+  }
+
+  onMetricChange = (value: string) => {
+    const { onChange, query, onRunQuery } = this.props;
+    onChange({ ...query, metric: value || '' });
+    onRunQuery();
+  };
+
+  onFunctionsChange = (value: any) => {
```
Comment (tilsche) [inline src/QueryEditor.tsx:30]:
```suggestion
  onFunctionsChange = (selectedFunctions:  Array<SelectableValue<string>>) => {
```

File: src/QueryEditor.tsx [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+    const { onChange, query, onRunQuery } = this.props;
+    onChange({ ...query, functions: value.map((val: SelectableValue) => val.value) });
```
Comment (tilsche) [inline src/QueryEditor.tsx:32]:
```suggestion
    onChange({ ...query, functions: selectedFunctions.map((val: SelectableValue<string>) => val.value) });
```

File: src/datasource.ts [hunk 5]
```diff
@@ -1,112 +1,135 @@
...
+          return {
+            status: 'failed',
+            message: response.status,
+          };
+        }
+      });
   }
 
-  mapToTextValue(result) {
+  mapToTextValue(result: any): MetricFindValue[] {
```
Comment (tilsche) [inline src/datasource.ts:100]: Replace with `mapToLabelValue`

**Review by tilsche — CHANGES_REQUESTED** [review]

File: src/QueryEditor.tsx (hunk 2 above)
Comment (kinnarr) [reply src/QueryEditor.tsx:129]: not required

File: src/datasource.ts (hunk 5 above)
Comment (kinnarr) [reply src/datasource.ts:100]: this function is used in `metricFindQuery` which requires `MetricFindValue` as return type

**Review by tilsche — APPROVED** [review]

**Pull request merged**